build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/types/DockerImage.java

```diff
@@ -22,51 +20,83 @@
      * Set the manifest and imagePath in case of the correct manifest.
-     *
-     * @param manifestPath
-     * @param candidateImagePath
-     * @param dependenciesClient
-     * @throws IOException
      */
-    private void checkAndSetManifestAndImagePathCandidates(String manifestPath, String candidateImagePath, ArtifactoryDependenciesClient dependenciesClient) throws IOException {
-        String candidateManifest = getManifestFromArtifactory(dependenciesClient, manifestPath);
+    private void checkAndSetManifestAndImagePathCandidates(String manifestPath, String candidateImagePath, ArtifactoryDependenciesClient dependenciesClient, boolean isLocalRepo) throws IOException {
+        String candidateManifest = getManifestFromArtifactory(dependenciesClient, manifestPath, isLocalRepo);
         String imageDigest = DockerUtils.getConfigDigest(candidateManifest);
         if (imageDigest.equals(imageId)) {
             manifest = candidateManifest;
             imagePath = candidateImagePath;
+            loadLayers();
         }
     }
 
-    private String getManifestFromArtifactory(ArtifactoryDependenciesClient dependenciesClient, String manifestPath) throws IOException {
+    /**
+     * Search for 'manifest.json' at 'manifestPath' in artifactory and return its content.
+     * If the target repository is not local type , fat-manifest can be found instead, which is a list of 'manifest.json' digest for one or more platforms.
+     * In order to find the right digest, we read and iterate on each digest and search for the one with the same os and arch.
+     * Using the correct digest from the fat-manifest, we are able to build the path toward manifest.json in Artifactory.
+     *
+     * @param dependenciesClient - Dependencies client builder.
+     * @param manifestPath       - Path to manifest in Artifactory.
+     * @param isLocalRepo        - Indicates if the search is against a local repository.
+     * @return The manifest content, otherwise throw an error.
+     * @throws IOException fail to search for manifest json in manifestPath.
+     */
+    private String getManifestFromArtifactory(ArtifactoryDependenciesClient dependenciesClient, String manifestPath, boolean isLocalRepo) throws IOException {
         HttpResponse res = null;
         try {
-            res = dependenciesClient.downloadArtifact(manifestPath);
-            return IOUtils.toString(res.getEntity().getContent());
+            res = dependenciesClient.downloadArtifact(manifestPath + "/manifest.json");
+            return IOUtils.toString(res.getEntity().getContent(), StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            if (isLocalRepo) {
+                throw e;
+            }
+            res = dependenciesClient.downloadArtifact(manifestPath + "/list.manifest.json");
+            String digestsFromFatManifest = DockerUtils.getImageDigestFromFatManifest(IOUtils.toString(res.getEntity().getContent()), this.os, this.architecture);
```

Comment:
1. You already have a dedicated method in DockerUtils to extract the string from the entity.
2. Unless you have local variables named os and architecture, there is no need to use the `this` keyword in methods that are no getters and setters.
```suggestion
            String digestsFromFatManifest = DockerUtils.getImageDigestFromFatManifest(entityToString(res.getEntity()), os, architecture);
```